generate-release-notes.sh

```diff
@@ -2,56 +2,112 @@
+	--al2-gpu-cuda-ver    (Optional) AL2 GPU CUDA version. If specified, then  --al2-gpu-nvidia optin is also required to be specified.
+	--al1-containerd-ver  (Optional) AL1 containerd version.
+	--exclude-ami         (Optional) comma separated list of AMI variants that are excluded in the release.
+
+Example:
+  $0 --al2-gpu-nvidia-ver 000.00.00 --al2-gpu-cuda-ver 00.0.0 --al1-containerd-ver 0.0.0 --exclude-ami al2023neu,al2inf
+EOF
 }
 
-# Parameters
-al2_gpu_nvidia_version=$1
-al2_gpu_cuda_version=$2
-al1_containerd_version=$3 # Optional, will include AL1 in the release notes if provided
+# Parses the options specified for the script.
+parse_args() {
+    while :; do
+        case $1 in
+        --al2-gpu-nvidia-ver)
+            AL2_GPU_NVIDIA_VERSION="$2"
+            shift
+            ;;
+        --al2-gpu-cuda-ver)
+            AL2_GPU_CUDA_VERSION="$2"
+            shift
+            ;;
+        --al1-containerd-ver)
+            AL1_CONTAINERD_VERSION="$2"
+            shift
+            ;;
+        --exclude-ami)
+            EXCLUDE_AMI="$2"
+            shift
+            ;;
+        --help)
+            usage
+            exit 0
+            ;;
+        --) # End of options.
+            shift
+            break
+            ;;
+        *) # Default case: No more options - break out of the loop.
+            break ;;
+        esac
+        shift
+    done
+}
 
-if [ "$al2_gpu_nvidia_version" == "" ]; then
-    echo "Error: AL2 GPU NVIDIA version is empty"
-    usage
-    exit 1
-fi
-if [ "$al2_gpu_cuda_version" == "" ]; then
-    echo "Error: AL2 GPU CUDA version is empty"
-    usage
-    exit 1
-fi
+parse_args "$@"
 
-# Read some information from pkrvars file
-readonly pkrvars="release.auto.pkrvars.hcl"
-readonly ami_version=$(cat $pkrvars | grep -w 'ami_version' | cut -d '"' -f2)
-readonly containerd_version_al2023=$(cat $pkrvars | grep -w 'containerd_version_al2023' | cut -d '"' -f2)
-readonly distribution_release_al2023=$(cat $pkrvars | grep -w 'distribution_release_al2023' | cut -d '"' -f2)
-readonly containerd_version=$(cat $pkrvars | grep -w 'containerd_version' | cut -d '"' -f2)
+# Checks if a given AMI variant is excluded in the release.
+is_ami_excluded() {
+    local ami="$1"
+    echo "$EXCLUDE_AMI" | grep -wq "$ami"
+}
 
-if [ -z "$ami_version" ]; then
-    echo "Error: AMI version was not found in $pkrvars"
-    exit 1
-fi
-if [ -z "$containerd_version_al2023" ]; then
-    echo "Error: Containerd version was not found for AL2023 in $pkrvars"
-    exit 1
-fi
-if [ -z "$distribution_release_al2023" ]; then
-    echo "Error: Distribution release version was not found for AL2023 in $pkrvars"
-    exit 1
-fi
-if [ -z "$containerd_version" ]; then
-    echo "Error: Containerd version was not found in $pkrvars"
+# Validates the options specified for the script.
+validate_args() {
+    if [ -z "$AL2_GPU_CUDA_VERSION" ] || [ -z "$AL2_GPU_NVIDIA_VERSION" ]; then
```

Comment:
Just curious, if both `AL2_GPU_CUDA_VERSION` and `AL2_GPU_NVIDIA_VERSION` and aren't empty, can al2gpu still be considered excluded? I'm assuming that we're typically setting these environment variables whenever we do need to release a new AL2 GPU AMI.

Reply:
Good question. If both AL2 GPU NVIDIA version and AL2 GPU CUDA version are passed in, but AL2 GPU is excluded from the release, then the passed in values for AL2 GPU NVIDIA version and AL2 GPU CUDA version are simply ignored since AL2 GPU is not being released.